### ISSUE TITLE
High memory usage while downloading large files

### ISSUE THREAD

**Issue by ShihabSoft** [bezzad/Downloader#134]
Hi there,

I've tried the
OnTheFlyDownload = true option, but this definition is not in the DownloadConfiguration class anymore.

I am seeing very high memory usage while downloading a humungous file. I've searched through all the issues but I don't find any other alternatives for this setting.

**Comment by ShihabSoft** [comment]
Update:

Found out that, when I enable ParallelDownload = true, memory usage shoots up and it consumes all the system RAM, disabling it fixes it.

**Comment by bezzad** [comment]
I released a new version (v3.0.5) to control memory buffering capacity. If you want to limit memory usage with the Downloader add the below config to your `DownloadConfiguration`:

```csharp
 MaximumMemoryBufferBytes = 1024 * 1024 * 50,  // release memory buffer after each 50 MB
```
